packages/form/addon/components/cf-field/input/file.js

```diff
@@ -4,64 +4,92 @@ import Component from "@glimmer/component";
 import { queryManager } from "ember-apollo-client";
 import fetch from "fetch";
 
-import removeAnswerMutation from "@projectcaluma/ember-form/gql/mutations/remove-answer.graphql";
-import getFileAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/fileanswer-info.graphql";
+import getFilesAnswerInfoQuery from "@projectcaluma/ember-form/gql/queries/filesanswer-info.graphql";
 
 export default class CfFieldInputFileComponent extends Component {
   @service intl;
 
   @queryManager apollo;
 
-  get downloadUrl() {
-    return this.args.field?.answer?.value?.downloadUrl;
+  get files() {
+    return this.args.field?.answer?.value;
   }
 
-  get downloadName() {
-    return this.args.field?.answer?.value?.name;
+  get allowedMimeTypes() {
+    // TODO: put this into a config env value
+    return ["image/png", "image/jpeg", "text/plain", "application/pdf"];
   }
 
   @action
-  async download() {
-    const { downloadUrl } = await this.apollo.query(
+  async download(fileName) {
+    if (!fileName) {
+      return;
+    }
+    const answers = await this.apollo.query(
       {
-        query: getFileAnswerInfoQuery,
+        query: getFilesAnswerInfoQuery,
         variables: { id: this.args.field.answer.raw.id },
         fetchPolicy: "network-only",
       },
-      "node.fileValue"
+      "node.value"
     );
 
+    const { downloadUrl } = answers.find((file) => file.name === fileName);
     if (downloadUrl) {
       window.open(downloadUrl, "_blank");
     }
   }
 
   @action
   async save({ target }) {
-    const file = target.files[0];
+    // store the old list of files
+    const fileList = this.files?.length ? this.files.slice() : [];
```

Comment:
I'd use destructuring for copying, something like
```suggestion
    const fileList = [...this.files] || []
```

Reply:
`this.files` can be `undefined` in some circumstances. As it isn't a `Iterable` then, it can not be destructured. So we will still need the `this.files?.length` check.

Reply:
Right, how about this then?
```js
const fileList = [...this.files || []] 
```

Reply:
Well this looks pretty!